README.md

```diff
@@ -18,6 +18,15 @@ of the pipeline description, and more.
  * **Mailing List**: image-builder@redhat.com
  * **Changelog**: <https://github.com/osbuild/osbuild/releases>
 
+### Principles
+
+1. OSBuild stages are never broken, only deprecated. The same manifest should always produce the same output.
+2. OSBuild stages should be explicit whenever possible instead of e.g. relying on the state of the tree.
+3. OSBuild's pipelines are independent, so the tree is expected to be empty at the beginning of each.
+4. Manifests are expected to be machine-generated, so OSBuild has no convenience functions to support manually created manifests.
+5. OSBuild has confinement around the build environment, but it is not a security feature.
+6. OSBuild must run on currently supported versions of RHEL, CentOS Stream and Fedora.
```

Comment:
(super nitpick) What I slightly dislike about the extended version of this sentence is that it now gives an impression that we care about RH/rpm stuff only. I.e. it could imply in a not charitable reading "it must run on this RH associated products but it's okay if it breaks on arch". But the real reason we had this originally was to say "we cannot use some modern python features or libraries because they are not part of RHEL". So maybe a more neutral way, something like `6. OSBuild must run on currently supported versions of our target distros` (or `... of our target distros (RHEL, Centos, Fedora, Arch`)?

Which makes me wonder if we should have a 
`7. OSBuild is distro/package manager agnostic and aspires to build rpm/arch/deb based distributions` 
(but that can be a followup, I understand we have not discussed this yet at all).

Reply:
Agree with this one.  I thought it would be nice to make it clearer what the implication is here, but it got a bit wordy and the principles are (and should be) generally terse.
So perhaps we could expand (maybe in a follow-up) with examples and implications in a separate section below the principles.  For example, this could be:

> OSBuild must run on all currently supported versions of our target distributions.

and the expanded description would then be:

> Requiring osbuild runs on all currently supported versions means that Python language features should only be used if they are supported by the oldest distribution version we support.  Additionally, differences in package versions used in stages should be considered and handled across the whole range of supported distro versions.

Reply:
"osbuild may only use Python language features supported by oldest target distribution."

Reply:
I like @bcl's version - it's very concise yet clear.

Reply:
Went with @bcl's suggestion